test/fluree/db/vector/index_test.clj

```diff
@@ -0,0 +1,192 @@
+(ns fluree.db.vector.index-test
+  (:require [clojure.test :refer :all]
+            [fluree.db.api :as fluree]
+            [fluree.db.constants :as const]
+            [fluree.db.json-ld.iri :as iri]
+            [fluree.db.test-utils :as test-utils]))
+
+(deftest ^:integration vector-index-search
+  (testing "Some vectors on a property can be flat-rank scored"
+    (let [conn   (test-utils/create-conn)
+          ledger @(fluree/create conn "vector-search")
+          db     @(fluree/stage
+                   (fluree/db ledger)
+                   {"@context" {"ex" "http://example.org/ns/"}
+                    "insert"
+                    [{"@id"     "ex:homer"
+                      "ex:xVec" {"@value" [0.6, 0.5]
+                                 "@type"  const/iri-vector}}
+                     {"@id"     "ex:lucy"
+                      "ex:xVec" "Not a Vector"} ;; <- a string value for ex:xVec
+                     {"@id"     "ex:bart"
+                      "ex:xVec" [{"@value" [0.1, 0.9]
+                                  "@type"  const/iri-vector}
+                                 {"@value" [0.2, 0.9]
+                                  "@type"  const/iri-vector}]}]})]
+
+      (testing "dot product scoring"
+        (let [query   {"@context" {"ex"   "http://example.org/ns/"
+                                   "fidx" iri/f-idx-ns}
+                       "select"   ["?x", "?score", "?vec"]
+                       "where"    [["graph"
+                                    "##Flatrank-DotProduct"
```

Comment:
What happens when we query datasets? For example, what if we'd like to query multiple dbs at once and we'd like to flat-rank search all of them together inside that query? What about if we'd like to search only the entries in a particular db inside of the broader query?